include/soci/type-conversion.h

```diff
@@ -120,6 +120,17 @@ class conversion_use_type
         //convert_to_base();
     }
 
+    conversion_use_type(T & value, indicator const & ind,
+            std::string const & name = std::string())
+        : use_type<base_type>(details::base_value_holder<T>::val_, ind, name)
+        , value_(value)
+        , ind_(ind)
```

Comment:
I must be missing something here... how can this ctor work when `ind` is a const reference and `ind_` a non-const one? AFAICS any attempt to use it should result in a compilation error.